### PR TITLE
0.8.1 SDK-1360

### PR DESCRIPTION
**Pull request by piyush-kukadiya**
- abstract out notification click callback for killed state.
- update example app

### CODE REVIEW

**Comment by darshanclevertap** [comment]
@piyush-kukadiya Abstraction looks fine, but we need to add documentation in case the client app uses a custom Application class. Add that to the README or Install/Integration doc file as well. Also, update CHANGELOG.

File: android/src/main/java/com/clevertap/react/CleverTapApplication.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        mReactInstanceManager.createReactContextInBackground();
+                    }
+                }
+
+            }
+        });
+
+    }
+
+    private WritableMap getWritableMapFromMap(Map<String, ? extends Object> var1) {
```
Comment (darshanclevertap) [inline android/src/main/java/com/clevertap/react/CleverTapApplication.java:67]: @piyush-kukadiya You can abstract this out to a Utils class and use it in `CleverTapModule.java` and `CleverTapApplication.java`

File: docs/integration.md [hunk 2]
```diff
@@ -71,5 +71,7 @@ Note: Need to use **@import CleverTapSDK;** instead of **#import <CleverTapSDK/C
         // ...
     }
     ```
+4. From clvevertap-react-native *v0.8.1* onwards developers can make their `Application` class extend `CleverTapApplication` to support Push Notification click callback out of the box and to register activity lifecycle events. Before v0.8.1 developers were forced to write logic for push click callback and register activity lifecycle to their `Application` class manually which is being abstract out in `CleverTapApplication` class.
```
Comment (darshanclevertap) [inline docs/integration.md:74]: Spelling error in CleverTap-React-Native. Make `v0.8.1` bold. And add code snippet for using `CleverTapApplication` class

File: android/src/main/java/com/clevertap/react/CleverTapApplication.java (hunk 1 above)
Comment (piyush-kukadiya) [reply android/src/main/java/com/clevertap/react/CleverTapApplication.java]: Done

File: docs/integration.md (hunk 2 above)
Comment (piyush-kukadiya) [reply docs/integration.md]: Done

**Pull request merged**